internal/agentlogger/agentlogger.go

```diff
@@ -0,0 +1,100 @@
+package agentlogger
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	agentapi "github.com/synadia-io/nex/internal/agent-api"
+	"sync/atomic"
+)
+
+const NexEventSourceNexAgent = "nex-agent"
+
+type AgentLogger struct {
+	entries chan *agentapi.LogEntry
+	nc      *nats.Conn
+	md      *agentapi.MachineMetadata
+	closing uint32
+}
+
+func (a *AgentLogger) LogEntry(entry *agentapi.LogEntry) {
+	if a.shuttingDown() {
+		return
+	}
+	a.entries <- entry
+}
+
+func (a *AgentLogger) LogMessage(lvl agentapi.LogLevel, msg string) {
+	a.LogEntry(&agentapi.LogEntry{
+		Source: NexEventSourceNexAgent,
+		Level:  lvl,
+		Text:   msg,
+	})
+}
+
+// Run pulls log entries off the channel and publish to the node host via internal NATS
+func (a *AgentLogger) Run() {
+	for !a.shuttingDown() {
+		entry := <-a.entries
+		err := a.publishLog(entry)
+		if err != nil {
+			continue
+		}
+	}
+}
+
+func (a *AgentLogger) publishLog(entry *agentapi.LogEntry) error {
+	bytes, err := json.Marshal(entry)
+	if err != nil {
+		return err
+	}
+
+	subject := fmt.Sprintf("agentint.%s.logs", *a.md.VmID)
+	err = a.nc.Publish(subject, bytes)
+	if err != nil {
+		return err
+	}
+
+	// TODO: why is the logger controlling the flush?
+	//  is this to avoiding dealing with draining?
+	err = a.nc.Flush()
```

Comment:
I need help here, any context about calling `Flush`? I am not sure why the logger would control that when the logger is capable to batch 64 entries in the channel